sdk/tables/azure-data-tables/inc/azure/data/tables/models.hpp

```diff
@@ -796,15 +786,12 @@ namespace Azure { namespace Data { namespace Tables {
       std::vector<Models::TableEntity> TableEntities;
 
       /**
-       * @brief Construct a new Query Entities Paged Response object.
+       * @brief Table Client.
        *
-       * @param tableClient Table client.
        */
-      QueryEntitiesPagedResponse(std::shared_ptr<TableClient> tableClient)
-          : m_tableClient(tableClient){};
+      std::shared_ptr<TableClient> m_tableClient;
```

Comment:
hmm, pubic?
Could you rather make `QueryTablesPagedResponse` a `friend` of `QueryEntitiesPagedResponse`, and modify this field then?

And I'm sorry, I think `private` constructor taking shared_ptr was a better idea, because when you declare a constructor, it removes all the public default auto-generated constructor, which is good for the customer, because they are not supposed to create instances of `QueryEntitiesPagedResponse`, other than get them as a copy via move constructor from calling `TableServiceClient::QueryTables()`.

BTW, for the constructor, one suggestion, to add std::move:
```cpp
#include <utility>

QueryEntitiesPagedResponse(std::shared_ptr<TableClient> tableClient)
  : m_tableClient(std::move(tableClient)){};
```